File: rb/features.rb
# N4chan SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module N4chanFeatures
  def self.make_feature(name)
    case name
    when "base"
      N4chanBaseFeature.new
    when "test"
      N4chanTestFeature.new
    else
      N4chanBaseFeature.new
    end
  end
end

File: py/features.py
# N4chan SDK feature factory

from feature.base_feature import N4chanBaseFeature
from feature.test_feature import N4chanTestFeature


def _make_feature(name):
    features = {
        "base": lambda: N4chanBaseFeature(),
        "test": lambda: N4chanTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
